scalyr_agent/platform_windows.py

```diff
@@ -446,10 +449,12 @@ def run_as_user(self, user_name, script_file, script_binary, script_arguments):
         @raise CannotExecuteAsUser: Indicates that the current process could not change the specified user for
             some reason to execute the script.
         """
-        if user_name != self.__local_administrators:
+        if (
+            user_name != self.__local_administrators
+            and user_name not in self.__all_local_administrator_names
+        ):
             raise CannotExecuteAsUser(
-                "The current Scalyr Agent implementation only supports running the agent as %s"
-                % self.__local_administrators
+                "The current Scalyr Agent implementation only supports running the agent as an Administrator account"
```

Comment:
I think it would be helpful to also include a list of all the admin account names in the error message.